src/services/data/handlers/sinkLegacyInsights.ts

```diff
@@ -0,0 +1,80 @@
+import { Handler } from "aws-lambda";
+import { decode } from "base-64";
+import * as os from "./../../../libs/opensearch-lib";
+import { KafkaEvent, KafkaRecord } from "shared-types";
+import { ErrorType, getTopic, logError } from "../libs/sink-lib";
+const osDomain = process.env.osDomain;
+if (!osDomain) {
+  throw new Error("Missing required environment variable(s)");
+}
+const index = "legacyinsights";
+
+export const handler: Handler<KafkaEvent> = async (event) => {
+  const loggableEvent = { ...event, records: "too large to display" };
+  const docs: any[] = [];
+  try {
+    for (const topicPartition of Object.keys(event.records)) {
+      const topic = getTopic(topicPartition);
+      switch (topic) {
+        case undefined:
+          logError({ type: ErrorType.BADTOPIC });
+          throw new Error();
+        case "aws.onemac.migration.cdc":
+          docs.push(
+            ...(await onemac(event.records[topicPartition], topicPartition)),
+          );
+          break;
+      }
+    }
+    try {
+      await os.bulkUpdateData(osDomain, index, docs);
+    } catch (error: any) {
+      logError({
+        type: ErrorType.BULKUPDATE,
+        metadata: { event: loggableEvent },
+      });
+      throw error;
+    }
+  } catch (error) {
+    logError({ type: ErrorType.UNKNOWN, metadata: { event: loggableEvent } });
+    throw error;
+  }
+};
+
+const onemac = async (kafkaRecords: KafkaRecord[], topicPartition: string) => {
```

Comment:
@13bfrancis TDD and TDR(efactoring) are great practices to prevent introducing bugs or losing functionality, definitely recommend at least getting those tests written and then pending remaining time, let your refactor scale accordingly. that's the approach i took when writing the other tests for shared-utils and i got a good improvement or two in :)